File: C#/0205. Isomorphic Strings.cs
public class Solution {
    public bool IsIsomorphic(string s, string t) {
        if(s.Length!=t.Length){return false;}
        IDictionary<char,char> sDict=new Dictionary<char,char>();
        HashSet<char> tSet=new HashSet<char>();
        
        bool flag=true;
        for(int i=0;i<s.Length;i++){
            if(!sDict.ContainsKey(s[i]) && tSet.Contains(t[i])){
                flag=false;
                break;
            }
            else if (!sDict.ContainsKey(s[i]) && !tSet.Contains(t[i])){
                sDict.Add(s[i],t[i]);
                tSet.Add(t[i]);
            }
            else if (sDict.ContainsKey(s[i]) && sDict[s[i]]!=t[i]){
                flag=false;
                break;
            }
        }
        return flag;
    }
}

File: archives/20190519python/0205. Isomorphic Strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        if len(s)!=len(t):
            return False
        dist_st={}
        used_word=set()
        for i in range(len(s)):
            if s[i] not in dist_st:
                if t[i] in used_word:
                    return False
                dist_st[s[i]]=t[i]
                used_word.add(t[i])
            else:
                if dist_st[s[i]]!=t[i]:
                    return False
        return True
